runtime-sdk/src/modules/core/test.rs

```diff
@@ -808,3 +818,64 @@ fn test_emit_events() {
     assert_eq!(TestEvent { i: 1 }, events[3], "expected events emitted");
     assert_eq!(TestEvent { i: 0 }, events[4], "expected events emitted");
 }
+
+/// Constructs a BTreeMap using a `btreemap! { key => value, ... }` syntax.
+macro_rules! btreemap {
+    // allow trailing comma
+    ( $($key:expr => $value:expr,)+ ) => (btreemap!($($key => $value),+));
+    ( $($key:expr => $value:expr),* ) => {
+        {
+            let mut m = BTreeMap::new();
+            $( m.insert($key.into(), $value); )*
+            m
+        }
+    };
+}
+
+#[test]
+fn test_module_info() {
+    use types::{MethodHandlerInfo, MethodHandlerKind};
+
+    let mut mock = mock::Mock::default();
+    let mut ctx = mock.create_ctx_for_runtime::<GasWasterRuntime>(Mode::CheckTx);
+
+    // Set bogus params on the core module; we want to see them reflected in response to the `runtime_info()` query.
+    let core_params = Parameters {
+        max_batch_gas: 123,
+        max_tx_signers: 4,
+        max_multisig_signers: 567,
+        ..Default::default()
+    };
+    Core::set_params(ctx.runtime_state(), core_params.clone());
+
+    let info = Core::query_runtime_info(&mut ctx, ()).unwrap();
+    assert_eq!(
+        info,
+        types::RuntimeInfoResponse {
+            runtime_version: Version::new(0, 0, 0),
+            state_version: 0,
+            modules: btreemap! {
+                "core" =>
+                    types::ModuleInfo {
+                        version: 1,
+                        params: cbor::to_vec(core_params),
+                        methods: Some(vec![
+                            MethodHandlerInfo { kind: MethodHandlerKind::Query, name: "core.EstimateGas".to_string() },
```

Comment:
Note: `MethodHandlerInfo` structs do not include arg types because that would require every arg to implement an introspection trait. Auto-deriving that for all possible types seems more work than it's worth.